src/main.ts

```diff
@@ -1,54 +1,68 @@
-import * as core from '@actions/core'
-import {context, getOctokit} from '@actions/github'
-import {OctokitResponse} from '@octokit/types'
+import * as core from '@actions/core';
+import {context, getOctokit} from '@actions/github';
+import {OctokitResponse} from '@octokit/types';
 
-import {AccessData, accessGroups} from './access'
+import {accessGroups} from './access';
+import {DataQuery, logAndExport} from './util';
 
 async function run(): Promise<void> {
-  try {
-    const user = context.actor // core.getInput('user') - future
-    const repo = context.payload.repository?.full_name
-    const token: string = core.getInput('github-token') // mirroring 'actions/github-script'
-
-    // const endpoint = core.getInput('endpoint')
-
-    const octokit = getOctokit(token)
-
-    const {data}: OctokitResponse<AccessData> = await octokit.request(
-      `
-      {
-        viewer {
-          isViewer
-          isEmployee
-          isHireable
-          isSiteAdmin
-          isGitHubStar
-          isBountyHunter
-          isCampusExpert
-          isSponsoringViewer
-          isDeveloperProgramMember
-        }
-      }
-      `
-    )
-
-    core.debug(`Access Data: ${JSON.stringify(data)}`)
-
-    const {groups} = accessGroups(
-      {
-        // endpoint
-        user,
-        repo
-      },
-      data
-    )
-
-    core.debug(`${user} has access to ${groups.join(', ')}`)
-
-    core.setOutput('groups', groups)
-  } catch (error) {
-    core.setFailed(error.message)
-  }
+	try {
+		console.log(`Actor: ${context.actor}`);
+		console.log(`Repo: ${JSON.stringify(context.repo)}`);
+
+		const {actor} = context; // Core.getInput('user') - future
+		const {owner, repo} = context.repo;
+
+		const token: string = core.getInput('github-token', {required: false});
+
+		const octokit = getOctokit(token, {
+			baseUrl: context.apiUrl,
+		});
+
+		const {data}: OctokitResponse<DataQuery> = await octokit.graphql(
+			`
+      query accessData($owner: String!, $repo: String!, $user: String!) {
+				user(login: $user) {
+					...Access
+				}
+				repository(owner: $owner, name: $repo) {
+					owner {
+						...Access
+					}
+					collaborators(query: $user) {
+						edges {
+							node {
+								...Access
+							}
+							permission
+						}
+					}
+				}
+			}
+			
+			fragment Access on User {
+				login
+				isViewer
+				isSiteAdmin
+				__typename
+			}
+      `,
+			{
+				owner,
+				repo,
+				user: actor,
```

Comment:
While the argument has been refactored / renamed or otherwise rewritten, the GraphQL client refuses to take the argument.
> That and the query checks are also incorrect, given that `~.viewer` no longer exists in this new query.